### PR TITLE
Popup-Circles

### PR DESCRIPTION
**Pull request by DanieleMaselli**
Awesome you want to make a PR  
please see [https://github.com/fabianmoronzirfas/gestalten-in-code/blob/master/.github/CONTRIBUTING.md](https://github.com/fabianmoronzirfas/gestalten-in-code/blob/master/.github/CONTRIBUTING.md) for further infos.

### CODE REVIEW

File: chance/p5js/Popup-Circles/index.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+layout: p5js
+author: Daniele Maselli
+author-url: https://github.com/DanieleMaselli 
+title: Popup-Circles 
+tags: /Chance/Repeat/Law of Similarity/Random
```
Comment (fabianmoronzirfas) [inline chance/p5js/Popup-Circles/index.md]: wrong format. Should be 

```yaml
tags:
  - chance
  - repeat
  - Law of Similarity
  - random
```

File: chance/p5js/Popup-Circles/index.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+layout: p5js
+author: Daniele Maselli
+author-url: https://github.com/DanieleMaselli 
+title: Popup-Circles
```
Comment (fabianmoronzirfas) [inline chance/p5js/Popup-Circles/index.md]: remove hyphen -

File: chance/p5js/Popup-Circles/index.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
+layout: p5js
+author: Daniele Maselli
+author-url: https://github.com/DanieleMaselli 
+title: Popup-Circles 
+tags: /Chance/Repeat/Law of Similarity/Random
+permalink: /chance/Popup-Circles/
```
Comment (fabianmoronzirfas) [inline chance/p5js/Popup-Circles/index.md]: lowercase please. This needs to be url safe

File: chance/p5js/Popup-Circles/index.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+summary: Crackling random circles. Psychedelic show!
```
Comment (fabianmoronzirfas) [inline chance/p5js/Popup-Circles/index.md]: Could it be even shorter?

File: chance/p5js/Popup-Circles/Index.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
+var initialOffset    = 10;
+var offset           = 25;
+var globalCircleSize = 15;
```
Comment (fabianmoronzirfas) [inline chance/p5js/Popup-Circles/Index.js]: Why the duplication of the `circleSize` it is not used anywhere else.

File: chance/p5js/Popup-Circles/Index.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+var col = {
```
Comment (fabianmoronzirfas) [inline chance/p5js/Popup-Circles/Index.js]: Why introducing color globally if it is overwritten locally with random values?

File: chance/p5js/Popup-Circles/Index.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        fill(col.r, col.g, col.b)
+
+        ellipse(x, y, circleSize, circleSize);
+        circleSize = Math.abs(randomGaussian(5, 10)) | 0;
+    }
+     
+    }
+}
+
+//play with strokeWeight :)
```
Comment (fabianmoronzirfas) [inline chance/p5js/Popup-Circles/Index.js]: unneeded comment.

File: chance/p5js/Popup-Circles/Index.js (hunk 6 above)
Comment (DanieleMaselli) [inline chance/p5js/Popup-Circles/Index.js]: not really random...from 100 to 180 and so on... 0 to 63....

Comment (fabianmoronzirfas) [inline chance/p5js/Popup-Circles/Index.js]: Yeah. But the variable is defined globally and then overwritten locally. Never used in another scope then draw.

Comment (DanieleMaselli) [inline chance/p5js/Popup-Circles/Index.js]: now it's in Draw

File: chance/p5js/Popup-Circles/index.md [hunk 8]
```diff
@@ -0,0 +1,9 @@
+---
+layout: p5js
+author: Daniele Maselli
+author-url: https://github.com/DanieleMaselli 
+title: Popup Circles 
+tags: /chance/repeat/Law of Similarity/random
```
Comment (fabianmoronzirfas) [inline chance/p5js/Popup-Circles/index.md]: tags are still wrong. Please take a look at https://github.com/fabianmoronzirfas/gestalten-in-code/wiki/Contributing-P5js-Sketches#markdown-and-front-matter

File: chance/p5js/Popup-Circles/index.md [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+permalink: /chance/Popup_Circles/
```
Comment (fabianmoronzirfas) [inline chance/p5js/Popup-Circles/index.md]: no Uppercae no underscore `/chance/popup-circles/`

File: chance/p5js/Popup-Circles/index.md [hunk 10]
```diff
@@ -0,0 +1,9 @@
...
+summary: Random crackling circles.
```
Comment (fabianmoronzirfas) [inline chance/p5js/Popup-Circles/index.md]: Making it even shorter was a joke. 🤦‍♂️

Comment (DanieleMaselli) [inline chance/p5js/Popup-Circles/index.md]: no jokes with pull requests..😅

File: chance/p5js/Popup-Circles/index.md [hunk 11]
```diff
@@ -0,0 +1,9 @@
+---
+layout: p5js
+author: Daniele Maselli
+author-url: https://github.com/DanieleMaselli 
+title: Popup Circles 
+tags: P5.js/chance/repeat/random/Law of Similarity
```
Comment (fabianmoronzirfas) [inline chance/p5js/Popup-Circles/index.md]: please fix the tags so we can merge this finally.

Comment (DanieleMaselli) [inline chance/p5js/Popup-Circles/index.md]: yes..sorry

**Comment by DanieleMaselli** [comment]
now it should be correct

File: chance/p5js/Popup-Circles/index.md [hunk 12]
```diff
@@ -0,0 +1,9 @@
+---
+layout: p5js
+author: Daniele Maselli
+author-url: https://github.com/DanieleMaselli 
+title: Popup Circles 
+tags: [P5.js, chance, repeat, random, Law of Similarity]
```
Comment (fabianmoronzirfas) [inline chance/p5js/Popup-Circles/index.md]: Nearly right. Did you take a look at the link? https://github.com/fabianmoronzirfas/gestalten-in-code/wiki/Contributing-P5js-Sketches#markdown-and-front-matter

We need to define the tags in a diffenret way then before. 🙇🏽‍♂️

Comment (DanieleMaselli) [inline chance/p5js/Popup-Circles/index.md]: sorry i though it was just the format wrong....but it is the repeat tag disturbing? or changed it completely

Comment (fabianmoronzirfas) [inline chance/p5js/Popup-Circles/index.md]: the repeat tag is totally okay.

File: chance/p5js/Popup-Circles/index.md [hunk 13]
```diff
@@ -3,7 +3,7 @@ layout: p5js
 author: Daniele Maselli
 author-url: https://github.com/DanieleMaselli 
 title: Popup Circles 
-tags: [P5.js, chance, repeat, random, Law of Similarity]
+tags: [P5.js, chance, random, form, Law of Connected Elements]
```
Comment (fabianmoronzirfas) [inline chance/p5js/Popup-Circles/index.md]: dear @DanieleMaselli i think we are misunderstanding us:  

```yaml
tags: 
  - P5.js
  - color
  - repeat
  - Law of Something
```

I'm totally okay with the repeat tag in there. The Laws you sort this in is a bit wired. Why connected elements? https://github.com/FH-Potsdam/laws-of-gestalt/blob/master/Law-of-Connected-Elements.md

Comment (DanieleMaselli) [inline chance/p5js/Popup-Circles/index.md]: but is the problem that i allready used a law of similarty in my past pull requests?

Comment (DanieleMaselli) [inline chance/p5js/Popup-Circles/index.md]: yes its a bit wired..i was a bit frustrating coz didnt know what u want

Comment (fabianmoronzirfas) [inline chance/p5js/Popup-Circles/index.md]: Well, the link to the wiki showed you an example how to format the frontmatter.

Comment (DanieleMaselli) [inline chance/p5js/Popup-Circles/index.md]: it looks now exacltly as your example..just chance instead of color

Comment (fabianmoronzirfas) [inline chance/p5js/Popup-Circles/index.md]: I'm not talking about the content all the time. I'm talking about how you format them:  

```yaml
tags: 
  - P5.js
  - color
  - repeat
  - Law of Something
```

Comment (DanieleMaselli) [inline chance/p5js/Popup-Circles/index.md]: same as everybody else does..or...

Comment (DanieleMaselli) [inline chance/p5js/Popup-Circles/index.md]: no? its the same as always did..+ now made some changes

**Comment by DanieleMaselli** [comment]
🎉

**Pull request merged**